src/main/java/com/powsybl/caseserver/service/CaseService.java

```diff
@@ -0,0 +1,170 @@
+import com.powsybl.caseserver.parsers.FileNameParsers;
+import com.powsybl.caseserver.repository.CaseMetadataEntity;
+import com.powsybl.caseserver.repository.CaseMetadataRepository;
+import com.powsybl.commons.datasource.DataSource;
+import com.powsybl.commons.datasource.DataSourceUtil;
+import com.powsybl.commons.datasource.MemDataSource;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Exporter;
+import com.powsybl.iidm.network.Importer;
+import com.powsybl.iidm.network.Network;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public interface CaseService {
+    default void validateCaseName(String caseName) {
+        Objects.requireNonNull(caseName);
+        if (!caseName.matches("[^<>:\"/|?*]+(\\.[\\w]+)")) {
+            throw CaseException.createIllegalCaseName(caseName);
+        }
+    }
+
+    default CaseInfos createInfos(String fileBaseName, UUID caseUuid, String format) {
+        FileNameParser parser = FileNameParsers.findParser(fileBaseName);
+        if (parser != null) {
+            Optional<? extends FileNameInfos> fileNameInfos = parser.parse(fileBaseName);
+            if (fileNameInfos.isPresent()) {
+                return CaseInfos.create(fileBaseName, caseUuid, format, fileNameInfos.get());
+            }
+        }
+        return CaseInfos.builder().name(fileBaseName).uuid(caseUuid).format(format).build();
+    }
+
+    default void createCaseMetadataEntity(UUID newCaseUuid, boolean withExpiration, boolean withIndexation, String originalFilename, String compressionFormat, CaseMetadataRepository caseMetadataRepository) {
+        Instant expirationTime = null;
+        if (withExpiration) {
+            expirationTime = Instant.now().plus(1, ChronoUnit.HOURS);
+        }
+        caseMetadataRepository.save(new CaseMetadataEntity(newCaseUuid, expirationTime, withIndexation, originalFilename, compressionFormat));
+    }
+
+    default void createCaseMetadataEntity(UUID newCaseUuid, boolean withExpiration, boolean withIndexation, CaseMetadataRepository caseMetadataRepository) {
+        createCaseMetadataEntity(newCaseUuid, withExpiration, withIndexation, null, null, caseMetadataRepository);
+    }
+
+    default List<CaseInfos> getMetadata(List<UUID> ids) {
+        List<CaseInfos> cases = new ArrayList<>();
+        ids.forEach(caseUuid -> {
+            CaseInfos caseInfos = getCaseInfos(caseUuid);
+            if (Objects.nonNull(caseInfos)) {
+                cases.add(caseInfos);
+            }
+        });
+        return cases;
+    }
+
+    default Importer getImporterOrThrowsException(Path caseFile, ComputationManager computationManager) {
```

Comment:
Like for caseMetadataService, don't add computation to the arguments, instead make a getComputationManager method

Reply:
done